src/openqaoa-core/problems/binpacking.py

```diff
@@ -0,0 +1,314 @@
+import numpy as np
+
+from typing import Union
+from docplex.mp.model import Model
+import matplotlib.pyplot as plt
+
+
+from .problem import Problem
+from .converters import FromDocplex2IsingModel
+
+
+class BinPacking(Problem):
+    """
+    Creates an instance of the bin packing problem.
+    https://en.wikipedia.org/wiki/Bin_packing_problem
+
+    Parameters
+    ----------
+    weights: List[int]
+        The weight of the items that must be placed in the bins.
+    weight_capacity: int
+        The maximum weight the bin can hold.
+    penalty: float
+        Penalty for the weight constraint.
+
+    Returns
+    -------
+        An instance of the bin packing problem.
+    """
+
+    __name__ = "bin_packing"
+
+    def __init__(
+        self,
+        weights: list = [],
+        weight_capacity: int = 0,
+        penalty: list = [],
+        n_bins: int = None,
+        simplifications=True,
+        method: str = "slack",
+    ):
+        # include_ineqs: True if including the inequalities
+        if method not in ["slack", "unbalanced"]:
+            raise ValueError(
+                f"The method '{method}' is not a valid method. Choose between 'slack' and 'unbalanced'"
+            )
+        for weight in weights:
+            if int(weight) != weight:
+                raise TypeError(f"The weights must be integer numbers. Format {type(weight)} found.")
+        if int(weight_capacity) != weight_capacity:
+            raise TypeError(f"The weight_capacity must be integer. Format {type(weight_capacity)} found.")
+        self.weights = weights
+        self.weight_capacity = weight_capacity
+        self.penalty = penalty
+        self.n_items = len(weights)
+        self.method = method
+        self.simplifications = simplifications
+        self.eq_constraints = {}
+        self.ineq_constraints = {}
+        if n_bins is None:
+            self.n_bins = self.n_items
+        else:
+            self.n_bins = n_bins
+
+        self.solution = self.solution_dict()
+        self.cplex_model = self.docplex_model()
+        self.n_vars = self.cplex_model.number_of_binary_variables
+
+    def solution_dict(self):
```

Comment:
Maybe we can include documentation on how the solution dictionary can be read by the user.
(i.e. what does x_0_1 in the key mean.)